src/fides/api/alembic/migrations/versions/5a8cee9c014c_privacy_preferences_v2_data.py

```diff
@@ -0,0 +1,377 @@
+"""
+
+CURRENT_PRIVACY_PREFERENCE_BASE_QUERY = """
+    SELECT
+        currentprivacypreference.id,
+        currentprivacypreference.preference,
+        currentprivacypreference.privacy_notice_history_id,
+        email_details.hashed_value as hashed_email, 
+        device_details.hashed_value as hashed_fides_user_device,
+        phone_details.hashed_value as hashed_phone_number,
+        email_details.encrypted_value as encrypted_email, 
+        device_details.encrypted_value as encrypted_device,
+        phone_details.encrypted_value as encrypted_phone,
+        currentprivacypreference.created_at,
+        currentprivacypreference.updated_at
+    FROM currentprivacypreference
+    LEFT OUTER JOIN providedidentity AS email_details ON email_details.field_name = 'email' AND email_details.id = currentprivacypreference.provided_identity_id
+    LEFT OUTER JOIN providedidentity AS phone_details ON phone_details.field_name = 'phone_number' AND phone_details.id = currentprivacypreference.provided_identity_id
+    LEFT OUTER JOIN providedidentity AS device_details ON device_details.field_name = 'fides_user_device_id' AND device_details.id = currentprivacypreference.fides_user_device_provided_identity_id
+    WHERE privacy_notice_history_id IS NOT NULL
+    ORDER BY created_at asc;
+"""
+
+
+LAST_SERVED_NOTICE_BASE_QUERY = """
+    SELECT
+        lastservednotice.id,
+        lastservednotice.privacy_notice_history_id,
+        email_details.hashed_value as hashed_email, 
+        device_details.hashed_value as hashed_fides_user_device,
+        phone_details.hashed_value as hashed_phone_number,
+        email_details.encrypted_value as encrypted_email, 
+        device_details.encrypted_value as encrypted_device,
+        phone_details.encrypted_value as encrypted_phone,
+        lastservednotice.created_at,
+        lastservednotice.updated_at
+    FROM lastservednotice
+    LEFT OUTER JOIN providedidentity AS email_details ON email_details.field_name = 'email' AND email_details.id = lastservednotice.provided_identity_id
+    LEFT OUTER JOIN providedidentity AS phone_details ON phone_details.field_name = 'phone_number' AND phone_details.id = lastservednotice.provided_identity_id
+    LEFT OUTER JOIN providedidentity AS device_details ON device_details.field_name = 'fides_user_device_id' AND device_details.id = lastservednotice.fides_user_device_provided_identity_id
+    WHERE privacy_notice_history_id IS NOT NULL
+    ORDER BY created_at asc;
+"""
+
+
+class CurrentMigrationType(Enum):
+    preferences = "preferences"
+    served = "served"
+
+
+encryptor = sqlalchemy_utils.types.encrypted.encrypted_type.StringEncryptedType(
+    String,
+    CONFIG.security.app_encryption_key,
+    AesGcmEngine,
+    "pkcs5",
+)
+
+decryptor = sqlalchemy_utils.types.encrypted.encrypted_type.StringEncryptedType(
+    JSONTypeOverride,
+    CONFIG.security.app_encryption_key,
+    AesGcmEngine,
+    "pkcs5",
+)
+
+
+def migrate_current_records(
```

Comment:
Migrating "current" privacy preferences and last notices served were challenging because the before/after have different sets of unique constraints.